frontend/src/main/scala/akkaviz/frontend/components/ReplTerminal.scala

```diff
@@ -0,0 +1,81 @@
+package akkaviz.frontend.components
+
+import akkaviz.frontend.terminal._
+import akkaviz.frontend.{DOMGlobalScope, FrontendUtil}
+import org.scalajs.dom.raw.{CloseEvent, Element, ErrorEvent, WebSocket}
+import org.scalajs.dom.{html, Event, MessageEvent}
+
+import scala.scalajs.js
+import scala.scalajs.js.typedarray.ArrayBuffer
+import scalatags.JsDom.all._
+
+class ReplTerminal extends Component {
+
+  private def options = js.Dynamic.literal(
+    cols = 120,
+    rows = 25
+  ).asInstanceOf[TerminalOptions]
+
+  private var ws: js.UndefOr[WebSocket] = js.undefined
+
+  private lazy val terminal = new Terminal(options)
+
+  private def setupWebsocket(): Unit = {
+
+    def writeToTerminal(s: String): Unit = {
+      terminal.write(s)
+    }
+
+    def decodeArrayBuffer(buffer: ArrayBuffer): String = {
+      DOMGlobalScope.ab2str(buffer)
+    }
+
+    def encodeArrayBuffer(str: String): ArrayBuffer = {
+      DOMGlobalScope.str2ab(str)
+    }
+
+    def disconnected(): Unit = {
+      terminal.off("data")
+      ws = js.undefined
+    }
+
+    val _ws = new WebSocket(FrontendUtil.webSocketUrl("repl"))
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Field name does not match the regular expression '^[a-z][A-Za-z0-9]*$'](https://www.codacy.com/app/lustefaniak/akka-viz/file/2101331309/issues/source?bid=3121282&fileBranchId=3142893#l42)